tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -74,40 +77,59 @@ private void Run(BsonDocument shared, BsonDocument test)
 
             _databaseName = shared["database_name"].AsString;
             _collectionName = shared["collection_name"].AsString;
 
+            KillAllSessions();
             DropCollection();
             CreateCollection();
             InsertData(shared);
 
-            var eventCapturer = new EventCapturer()
-                .Capture<CommandStartedEvent>(e => !__commandsToNotCapture.Contains(e.CommandName));
+            using (ConfigureFailPoint(test))
+            {
+                var eventCapturer = new EventCapturer()
+                    .Capture<CommandStartedEvent>(e => !__commandsToNotCapture.Contains(e.CommandName));
 
-            Dictionary<string, BsonValue> sessionIdMap;
+                Dictionary<string, BsonValue> sessionIdMap;
 
-            using (var client = CreateDisposableClient(test, eventCapturer))
-            using (var session0 = StartSession(client, test, "session0"))
-            using (var session1 = StartSession(client, test, "session1"))
-            {
-                var objectMap = new Dictionary<string, object>
+                using (var client = CreateDisposableClient(test, eventCapturer))
+                using (var session0 = StartSession(client, test, "session0"))
+                using (var session1 = StartSession(client, test, "session1"))
                 {
-                    { "session0", session0 },
-                    { "session1", session1 }
-                };
-                sessionIdMap = new Dictionary<string, BsonValue>
-                {
-                    { "session0", session0.ServerSession.Id },
-                    { "session1", session1.ServerSession.Id }
-                };
+                    var objectMap = new Dictionary<string, object>
+                    {
+                        { "session0", session0 },
+                        { "session1", session1 }
+                    };
+                    sessionIdMap = new Dictionary<string, BsonValue>
+                    {
+                        { "session0", session0.ServerSession.Id },
+                        { "session1", session1.ServerSession.Id }
+                    };
+
+                    ExecuteOperations(client, objectMap, test);
+                }
 
-                ExecuteOperations(client, objectMap, test);
+                AssertEvents(eventCapturer, test, sessionIdMap);
+                AssertOutcome(test);
             }
+        }
 
-            AssertEvents(eventCapturer, test, sessionIdMap);
-            AssertOutcome(test);
+        private void KillAllSessions()
+        {
+            var client = DriverTestConfiguration.Client;
+            var adminDatabase = client.GetDatabase("admin");
+            var command = BsonDocument.Parse("{ killAllSessions : [] }");
+            try
+            {
+                adminDatabase.RunCommand<BsonDocument>(command);
+            }
+            catch (MongoCommandException)
+            {
+                // ignore { ok : 0 } exceptions
```

Comment:
Should this catch iff it's an { ok: 0 } exception?

Reply:
I changed the comment to avoid confusion.